pkg/pod/status.go

```diff
@@ -394,10 +383,42 @@ func getFailureMessage(logger *zap.SugaredLogger, pod *corev1.Pod) string {
 		}
 	}
 
+	// Next, try to surface an error about the actual init container that failed.
+	for _, status := range pod.Status.InitContainerStatuses {
+		if msg := extractContainerFailureMessage(logger, status, pod.ObjectMeta); len(msg) > 0 {
+			return fmt.Sprintf("init container failed, %s", msg)
+		}
+	}
+
```

Comment:
Do we want to move this code block before surfacing actual build step errors? As the init container runs before those actual build steps?

Reply:
good idea, i have adjusted.